runtime/autoload/man.vim

```diff
@@ -118,15 +118,12 @@ endfunction
 " Expects a string like 'access' or 'access(2)'.
 function s:parse_page_and_section(sect, str) abort
   try
-    let save_isk = &iskeyword
-    setlocal iskeyword-=(,)
-    let page = substitute(a:str, '(*\(\k\+\).*', '\1', '')
-    let sect = substitute(a:str, '\(\k\+\)(\([^()]*\)).*', '\2', '')
+    let page = substitute(a:str, '\([-.[:alnum:]_]\+\).*', '\1', '')
+    let sect = substitute(a:str, '\([-.[:alnum:]_]\+\)(\([^()]*\)).*', '\2', '')
     if sect == page || -1 == match(sect, '^[0-9 ]\+$')
```

Comment:
By the way, what with sections like `{number}p` and `{letter}`?

I have e.g. `man 3p printf` (“POSIX programmer’s manual”) and `man 3 printf` (“linux programmer’s manual”) describing C printf function, `man 1p printf` (“POSIX programmer’s manual”) and `man 1 printf` (“user commands”: this one is from coreutils) describing `/usr/bin/printf`. This is Gentoo, sys-apps/man-pages-posix package, I do not know whether such thing exists in your distribution.

I also have `n` section: this is from tcl, lapack, blas; all work. And not working `m` section from motif and `x` section from libcmis.

I actually suspect that section may be *any* string that can be a file name.